src/yarp.c

```diff
@@ -12137,6 +12137,9 @@ parse_expression_prefix(yp_parser_t *parser, yp_binding_power_t binding_power) {
             yp_statements_node_t *statements = parse_statements(parser, YP_CONTEXT_PREEXE);
 
             expect1(parser, YP_TOKEN_BRACE_RIGHT, YP_ERR_BEGIN_UPCASE_TERM);
+            if ((parser->current_context->context != YP_CONTEXT_MAIN) && (parser->current_context->context != YP_CONTEXT_PREEXE)) {
```

Comment:
Can you extract out a variable for the `context` so we're not accessing it twice?